### PR TITLE
PAY-3573 : Liberata Pull APIs are unhooked from Internal Payment Swagger

### PR DESCRIPTION
**Pull request by AyanDuttaHMCTS**
### JIRA link (if applicable) ###
PAY-3573 : Liberata Pull APIs are unhooked from Internal Payment Swagger

### Change description ###

PAY-3573 : Liberata Pull APIs are unhooked from Internal Payment Swagger

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: api/src/main/java/uk/gov/hmcts/payment/api/controllers/liberata/LiberataPullApiController.java [hunk 1]
```diff
@@ -0,0 +1,236 @@
+package uk.gov.hmcts.payment.api.controllers.liberata;
```
Comment (YasharRahvar) [inline api/src/main/java/uk/gov/hmcts/payment/api/controllers/liberata/LiberataPullApiController.java:1]: We should not name our packages with 3rd party names. Please rename this.

File: api/src/main/java/uk/gov/hmcts/payment/api/controllers/liberata/LiberataPullApiController.java [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+import uk.gov.hmcts.payment.api.validators.PaymentValidator;
+
+import java.math.BigDecimal;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@RestController
+@Api(tags = {"Payment"})
+@SwaggerDefinition(tags = {@Tag(name = "LiberataPullApiController", description = "Payment REST API")})
+public class LiberataPullApiController {
```
Comment (YasharRahvar) [inline api/src/main/java/uk/gov/hmcts/payment/api/controllers/liberata/LiberataPullApiController.java:37]: Not the best name for a class. Please do not use 3rd party names in the code.

**Review by YasharRahvar — CHANGES_REQUESTED** [review]

File: api/src/main/java/uk/gov/hmcts/payment/api/controllers/liberata/LiberataPullApiController.java (hunk 1 above)
Comment (AyanDuttaHMCTS) [reply api/src/main/java/uk/gov/hmcts/payment/api/controllers/liberata/LiberataPullApiController.java]: Done

File: api/src/main/java/uk/gov/hmcts/payment/api/controllers/liberata/LiberataPullApiController.java (hunk 2 above)
Comment (AyanDuttaHMCTS) [reply api/src/main/java/uk/gov/hmcts/payment/api/controllers/liberata/LiberataPullApiController.java]: Done

**Review by YasharRahvar — APPROVED** [review]

File: api/src/main/java/uk/gov/hmcts/payment/api/controllers/payment/ReconciliationApiController.java [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+        return new PaymentsResponse(paymentDtos);
+    }
+
+    @ApiOperation(value = "Update payment status by payment reference", notes = "Update payment status by payment reference")
+    @ApiResponses(value = {
+        @ApiResponse(code = 204, message = "No content"),
+        @ApiResponse(code = 404, message = "Payment not found")
+    })
+    @PaymentExternalAPI
+    @PatchMapping("/payments/{reference}/status/{status}")
```
Comment (aravindan89) [inline api/src/main/java/uk/gov/hmcts/payment/api/controllers/payment/ReconciliationApiController.java:128]: I understand that we are moving **/payments** which is the end point used by liberata to pull the data to ReconciliationApiController. But why we need to move **/payments/{reference}/status/{status}** to ReconciliationApiController?

Comment (AyanDuttaHMCTS) [reply api/src/main/java/uk/gov/hmcts/payment/api/controllers/payment/ReconciliationApiController.java:128]: This is another external API which might get noticed in future ITHC reviews and we need to refactor, so we have moved to separate external API Controller.

**Review by aravindan89 — APPROVED** [review]

**Pull request closed without merging**